platform/darwin/src/MGLGeoJSONSource.mm

```diff
@@ -102,6 +103,42 @@ - (void)validateValue:(id)value
     }
 }
 
+- (void)setGeoJSONData:(NSData *)geoJSONData
+{
+    _geoJSONData = geoJSONData;
+    
+    NSString *string = [[NSString alloc] initWithData:_geoJSONData encoding:NSUTF8StringEncoding];
+    const auto geojson = mapbox::geojson::parse(string.UTF8String).get<mapbox::geojson::feature_collection>();
+    
+    const auto geoJSONSource = reinterpret_cast<mbgl::style::GeoJSONSource *>(self.source);
```

Comment:
These `reinterpret_cast`s would be unnecessary if this class instead has a class extension that redefines the `source` property to be of type `mbgl::style::GeoJSONSource *`. We follow this covariant pattern in a few places, notably the style layer classes.